execution/types/receipt.go

```diff
@@ -576,6 +576,11 @@ func (e receiptEncoder69) EncodeRLP(w io.Writer) error { return e.r.EncodeRLP69(
 func (rs Receipts) EncodeRLP69(w io.Writer) error {
 	encs := make([]receiptEncoder69, len(rs))
 	for i := range rs {
+		// state sync receipts have CumulativeGasUsed == 0 and Type == 0
+		// to avoid ambiguity, we force Type to 0 if CumulativeGasUsed == 0
+		if rs[i].CumulativeGasUsed == 0 {
```

Comment:
We derive cumulative gas so it may not be zero going forward. It's better to have some different indicator for identifying state-sync transactions.

Reply:
I think you can borrow the logic you're using in `ReadStateSyncReceiptByHash` function here to identify if a receipt is state-sync or not.

Reply:
Done here ff196b605596acb39d4efa136b0d1ffcc2197244
Please double check
Thanks